lib/util.js

```diff
@@ -346,27 +347,38 @@ function _fetch_rss_json (feed_url) {
     });
 }
 
-function _fetch_rss_page (feed, page, items, max_items, oldest_date) {
-    let feed_url, is_paginated;
+function _fetch_rss_page (feed, page, items, all_items, max_items, oldest_date) {
+    let feed_obj, is_paginated;
     if (typeof feed === 'function') {
         is_paginated = true;
-        feed_url = feed(page);
+        feed_obj = feed(page);
     } else {
         is_paginated = false;
-        feed_url = feed;
+        feed_obj = feed;
+    }
+
+    let feed_url = feed_obj, is_array = false;
+    if (Array.isArray(feed_obj)) {
+        is_array = true;
+        feed_url = feed_obj.shift();
     }
 
     return _fetch_rss_json(feed_url).then(feed_json => {
-        const limited_items = feed_json.items.slice(0, max_items - items.length);
-        const recent_enough_items = limited_items.filter(article => article.pubdate >= oldest_date);
-        const new_items = items.concat(recent_enough_items);
+        const recent_enough_items = feed_json.items.filter(item => item.pubdate >= oldest_date && !all_items.has(item.link));
+        const limited_items = recent_enough_items.slice(0, max_items - items.length);
```

Comment:
Oh yes! we were cutting max items, then filtering by date. Should be the other way round as you did.